### ISSUE TITLE
Broken xmod files generated

### ISSUE THREAD

**Issue by m214089** [omni-compiler/xcodeml-tools#160]
Hi,
we are using claw-compiler, which internally uses the xcodeml-tools. 
Now we hit a problem of incorrect xml in xmod files. Furthermore, this 
seems to be a regression.

If we use revision  772cf79077ac5f04cfc204fc7e0b53755a362d8b an xmod file containing
```
mtime_hl.xmod:          <name><</name>
mtime_hl.xmod:          <name><=</name>
mtime_hl.xmod:          <name><</name>
mtime_hl.xmod:          <name><=</name>
```
is generated, which is mal-formed xml. And when the omni-compiler tries to read this the compiler core dumps.

If we use revision 8d782adcd2fc521911c927540682ce6c0fa620bc the respective entries in the xmod file contain

```
<name>.lt.</name>
<name>.le.</name>
<name>.lt.</name>
<name>.le.</name>
```
which get processed without a problem afterwards.

Is it possible to get a fix?

Thanks a lot,
Luis

**Comment by m214089** [comment]
I did a bisect session: the last good revision is 0773953e ...

**Comment by clementval** [comment]
Fix in #162
